lib/compat/wordpress-6.0/class-wp-theme-json-gutenberg.php

```diff
@@ -119,6 +165,69 @@ protected function get_block_classes( $style_nodes ) {
 		return $block_rules;
 	}
 
+	/**
+	 * Given a styles array, it extracts the style properties
+	 * and adds them to the $declarations array following the format:
+	 *
+	 * ```php
+	 * array(
+	 *   'name'  => 'property_name',
+	 *   'value' => 'property_value,
+	 * )
+	 * ```
+	 *
+	 * @param array $styles Styles to process.
+	 * @param array $settings Theme settings.
+	 * @param array $properties Properties metadata.
+	 * @param string|null $selector Current selector.
+	 * @return array Returns the modified $declarations.
+	 */
+	protected static function compute_style_properties( $styles, $settings = array(), $properties = null, $selector = null ) {
+		if ( null === $properties ) {
+			$properties = static::PROPERTIES_METADATA;
+		}
+
+		$declarations = array();
+		if ( empty( $styles ) ) {
+			return $declarations;
+		}
+
+		foreach ( $properties as $css_property => $value_path ) {
+			// Some styles such as blockGap are only meant to be available at the top level (ROOT_BLOCK_SELECTOR),
+			// hence we only output styles at the top level.
+			if ( 'top' === _wp_array_get( self::VALID_STYLES, array( $value_path[0], $value_path[1] ), null ) && static::ROOT_BLOCK_SELECTOR !== $selector ) {
+				continue;
```

Comment:
I'm suspect of this. It seems fragile, only in the sense that I'm not aware of any side-effects. 

Needs lots of testing.